src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCM.java

```diff
@@ -348,23 +350,24 @@ public ListBoxModel doFillGitToolItems() {
         @POST
         public ListBoxModel doFillServerIdItems(@QueryParameter String serverId) {
             Jenkins.get().checkPermission(Permission.CONFIGURE);
-            List<BitbucketServerConfiguration> serverList =
-                    new BitbucketPluginConfiguration().getServerList();
+            //Filtered to only include valid server configurations
+            List<BitbucketServerConfiguration> serverList = bitbucketPluginConfiguration.getServerList();
             StandardListBoxModel model = new StandardListBoxModel();
-            if (isBlank(serverId) || serverList.isEmpty()) {
+            List<Option> options =
+                    bitbucketPluginConfiguration.getServerList()
+                            .stream()
+                            .filter(server -> server.getId().equals(serverId) ||
+                                              server.validate().kind == FormValidation.Kind.OK)
+                            .map(server ->
+                                    new Option(
+                                            server.getServerName(),
+                                            server.getId(),
+                                            server.getId().equals(serverId)))
+                            .collect(Collectors.toList());
+            model.addAll(options);
```

Comment:
Another option instead of creating the `model` and a new `List` just to add all the items into the model is to collect them all directly to the model with
```java
StandardListBoxModel model =
...
.collect(toCollection(StandardListBoxModel::new));
```
Note that you'll have to update your `if` to check `model.isEmpty()` instead of `options`